### PR TITLE
Silence some warnings

### PR DESCRIPTION
**Pull request by hcorion**
This touches all different parts of the code base, so I'm requesting review from @Nekotekina and @kd-11 , as some assumptions I made may be incorrect, and some of the errors may provide wrong/not enough information.

### CODE REVIEW

File: Utilities/Log.cpp [hunk 1]
```diff
@@ -41,9 +41,8 @@ void fmt_class_string<logs::level>::format(std::string& out, u64 arg)
 		case logs::level::warning: return "Warning";
 		case logs::level::notice: return "Notice";
 		case logs::level::trace: return "Trace";
+		case logs::level::_uninit: return unknown;
 		}
-
```
Comment (Nekotekina) [inline Utilities/Log.cpp]: It shouldn't be abandoned in most cases

File: rpcs3/Emu/Cell/lv2/lv2.cpp [hunk 2]
```diff
@@ -969,9 +969,8 @@ void fmt_class_string<CellError>::format(std::string& out, u64 arg)
 		STR_CASE(CELL_ESDKVER);
 		STR_CASE(CELL_ENOLICDISC);
 		STR_CASE(CELL_ENOLICENT);
+		default: return unknown;
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/lv2.cpp]: That was the point: see the warning and add unimplemented case.

File: Utilities/Log.cpp (hunk 1 above)
Comment (hcorion) [reply Utilities/Log.cpp]: Addresed

File: rpcs3/Emu/Cell/lv2/lv2.cpp (hunk 2 above)
Comment (hcorion) [reply rpcs3/Emu/Cell/lv2/lv2.cpp]: Ok, reverted in hopes that it gets implement soon :+1:

File: rpcs3/Emu/RSX/GL/GLHelpers.cpp [hunk 3]
```diff
@@ -22,8 +22,8 @@ namespace gl
 		case rsx::primitive_type::quads: return GL_TRIANGLES;
 		case rsx::primitive_type::quad_strip: return GL_TRIANGLE_STRIP;
 		case rsx::primitive_type::polygon: return GL_TRIANGLES;
+		default: fmt::throw_exception("unknow primitive type" HERE);
```
Comment (OverlordCW) [inline rpcs3/Emu/RSX/GL/GLHelpers.cpp]: unknowN

Comment (kd-11) [reply rpcs3/Emu/RSX/GL/GLHelpers.cpp]: Put all throws on their own line. breakpoints are the only way to obtain backtraces to identify the culprit

Comment (hcorion) [reply rpcs3/Emu/RSX/GL/GLHelpers.cpp]: Good catch, thanks @OverlordCW .

Should be addressed @kd-11 .

File: Utilities/Config.cpp [hunk 4]
```diff
@@ -202,6 +202,7 @@ void cfg::encode(YAML::Emitter& out, const cfg::_base& rhs)
 		out << YAML::EndMap;
 		return;
 	}
+	default: LOG_ERROR(GENERAL, "Attempting to cfg encode an unimplemented type.");
```
Comment (kd-11) [inline Utilities/Config.cpp]: default on newline

File: rpcs3/Emu/RSX/GL/GLHelpers.h [hunk 5]
```diff
@@ -1379,9 +1379,8 @@ namespace gl
 			case internal_format::compressed_rgba_s3tc_dxt3:
 			case internal_format::compressed_rgba_s3tc_dxt5:
 				return true;
+			default: return false;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/GL/GLHelpers.h]: same here

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 6]
```diff
@@ -1071,9 +1071,8 @@ namespace rsx
 			case rsx::vertex_base_type::s32k:
 			case rsx::vertex_base_type::ub256:
 				return true;
+			default: return false;
```
Comment (kd-11) [inline rpcs3/Emu/RSX/RSXThread.cpp]: and here

File: rpcs3/basic_mouse_handler.cpp [hunk 7]
```diff
@@ -55,6 +55,7 @@ bool basic_mouse_handler::eventFilter(QObject* target, QEvent* ev)
 		case QEvent::Wheel:
 			MouseScroll(static_cast<QWheelEvent*>(ev));
 			break;
+		default: return false;
```
Comment (kd-11) [inline rpcs3/basic_mouse_handler.cpp]: again

**Comment by hcorion** [comment]
Superceded by #3806

**Pull request closed without merging**